src/fs.rs

```diff
@@ -579,29 +619,10 @@ impl<IO: Read + Write + Seek, TP, OCC> FileSystem<IO, TP, OCC> {
         Ok(())
     }
 
-    pub(crate) fn set_dirty_flag(&self, dirty: bool) -> Result<(), IO::Error> {
-        // Do not overwrite flags read from BPB on mount
-        let mut flags = self.bpb.status_flags();
-        flags.dirty |= dirty;
-        // Check if flags has changed
-        let current_flags = self.current_status_flags.get();
-        if flags == current_flags {
-            // Nothing to do
-            return Ok(());
-        }
-        let encoded = flags.encode();
-        // Note: only one field is written to avoid rewriting entire boot-sector which could be dangerous
-        // Compute reserver_1 field offset and write new flags
-        let offset = if self.fat_type() == FatType::Fat32 {
-            0x041
-        } else {
-            0x025
-        };
+    #[inline]
+    pub(crate) fn set_dirty_flag(&self, dirty: bool) -> Result<(), Error<IO::Error>> {
         let mut disk = self.disk.borrow_mut();
-        disk.seek(io::SeekFrom::Start(offset))?;
-        disk.write_u8(encoded)?;
-        self.current_status_flags.set(flags);
-        Ok(())
+        set_dirty_flag(&mut *disk, self.fat_type(), &self.current_status_flags, dirty)
```

Comment:
why adding this new free `set_dirty_flag` function? AFAIK it is used only in this place.